### PR TITLE
feat: ✨ Add DFS content

### LINKED ISSUE #17: DFS
- [x] Pseudocode
- [x] Runtimes
- [x] Seen/done inequalities
- [x] Applications

### PR DESCRIPTION
**Pull request by pumbas600**
Closes #17

### CODE REVIEW

File: pages/searching/dfs.tsx [hunk 1]
```diff
@@ -1,10 +1,129 @@
...
+              <StyledCell>Θ(n + e) where e is the number of edges</StyledCell>
+            </TableRow>
+            <TableRow>
+              <StyledCell>Adjacency Matrix</StyledCell>
+              <StyledCell>Θ(n) per vertex for n vertices = Θ(n²)</StyledCell>
+            </TableRow>
+          </TableBody>
+        </StyledTable>
+      <Heading title="Types of Arcs" />
+      Let (v, w) be an arc in a digraph traversed by DFS.
```
Comment (pumbas600) [inline pages/searching/dfs.tsx:94]: If you wrap this in a `<Typography>` it should make it feel slightly further away from the title

File: pages/searching/dfs.tsx [hunk 2]
```diff
@@ -1,10 +1,129 @@
...
+              <StyledCell>seen[w] {"<"} seen[v] {"<"} done[v] {"<"} done[w]</StyledCell>
+            </TableRow>
+            <TableRow>
+              <StyledCell>Cross Arc</StyledCell>
+              <StyledCell>seen[w] {"<"} done[w] {"<"} seen[v] {"<"} done[v]</StyledCell>
+            </TableRow>
+          </TableBody>
+        </StyledTable>
+        <Heading title="Determining Ancestors Using DFS" />
+        Let v and w be nodes in a search forest F obtained by performing DFS. Additionally suppose that seen[v] {"<"} seen[w]
```
Comment (pumbas600) [inline pages/searching/dfs.tsx:122]: Could you wrap this in a `Typography` too.

**Review by pumbas600 — COMMENTED** [review]
Can you just quickly slap full stops at the end of all the sentences? Other than that, this is stunning :)

File: pages/searching/dfs.tsx (hunk 2 above)
Comment (atharvaarankalle) [reply pages/searching/dfs.tsx]: All done! :)

**Pull request merged**